synapse/storage/database.py

```diff
@@ -88,13 +88,18 @@ def make_pool(
     """Get the connection pool for the database.
     """
 
+    # By default enable `cp_reconnect`. We need to fiddle with db_args in case
+    # someone has explicitly set `cp_reconnect`.
+    db_args = dict(db_config.config.get("args", {}))
```

Comment:
What's the reason for the explicit `dict` cast? In case users put a list here instead or something?

Reply:
To take a copy so that we don't modify `db_config`